### PR TITLE
build: make uninstall related files of runtime

### LINKED ISSUE #5225: request help: support `make uninstall`
### Issue description

What about support `make uninstall` as a reverse of `make install`?
I assume this requires a little refactor in the Makefile and can be a useful improvement.

### Environment

- apisix version (cmd: `apisix version`): commit 2b23907

### PR DESCRIPTION
**Pull request by kwanhur**
### What this PR does / why we need it:

`make uninstall` on no need of the runtime

Fix #5225 

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/how-to-build.md [hunk 1]
```diff
@@ -117,6 +117,15 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
...
         Solution: set `LDAP_DIR` with `luarocks config` manually, for example `luarocks config variables.LDAP_DIR /usr/local/opt/openldap/`
 
+5. If you have no more need of the Apache APISIX runtime, uninstall it.
+
+```shell
+  # Uninstall apisix command
+  $ make uninstall
+```
+
+  Attention please, this operation will totally **remove** the related files.
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:127]:
```suggestion
Attention please, this operation will totally **remove** the related files.
```

File: docs/en/latest/how-to-build.md [hunk 2]
```diff
@@ -117,6 +117,15 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
 
         Solution: set `LDAP_DIR` with `luarocks config` manually, for example `luarocks config variables.LDAP_DIR /usr/local/opt/openldap/`
 
+5. If you have no more need of the Apache APISIX runtime, uninstall it.
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:120]:
```suggestion
5. If you have no more need for the Apache APISIX runtime, you could uninstall it like this.
```

**Review by spacewander — APPROVED** [review]

File: docs/en/latest/how-to-build.md (hunk 1 above)
Comment (kwanhur) [reply docs/en/latest/how-to-build.md:127]: line#127 aligns at line#97, so it starts with two space.

**Review by nic-chen — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -35,6 +35,7 @@ ENV_APISIX             ?= $(CURDIR)/bin/apisix
 ENV_GIT                ?= git
 ENV_TAR                ?= tar
 ENV_INSTALL            ?= install
+ENV_RM                 ?= rm
```
Comment (leslie-tsang) [inline Makefile:38]:
```suggestion
ENV_RM                 ?= rm -vf
```
let's extraction of public parts here, :)

File: Makefile [hunk 4]
```diff
@@ -340,6 +341,16 @@ install: runtime
 	$(ENV_INSTALL) apisix/plugins/slslog/*.lua $(ENV_INST_LUADIR)/apisix/plugins/slslog/
 
 
+### uninstall : Uninstall the apisix
+.PHONY: uninstall
+uninstall:
+	@$(call func_echo_status, "$@ -> [ Start ]")
+	$(ENV_RM) -vfr /usr/local/apisix
+	$(ENV_RM) -vfr $(ENV_INST_LUADIR)/apisix
+	$(ENV_RM) -vf $(ENV_INST_BINDIR)/apisix
```
Comment (leslie-tsang) [inline Makefile:350]:
```suggestion
	$(ENV_RM) -r /usr/local/apisix
	$(ENV_RM) -r $(ENV_INST_LUADIR)/apisix
	$(ENV_RM) $(ENV_INST_BINDIR)/apisix
```

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**